### PR TITLE
Extend domain with a name

### LINKED ISSUE #211: Extend domain with a name
It would be nice to refer to the specific domain in messages/exceptions.
Extend the yaml file with a name field and the Domain class with a get_name() method.

### PR DESCRIPTION
**Pull request by NoB0**
Add the field `name` to domain configuration. Implement `get_name` and associated test.

Fixes #211

### CODE REVIEW

File: dialoguekit/core/domain.py [hunk 1]
```diff
@@ -10,20 +10,30 @@ class Domain:
     def __init__(self, config_file: str) -> None:
         """Represents domain knowledge.
 
+        The YAML configuration file should contain at least the fields name and
+        slot_names.
```
Comment (kbalog) [inline dialoguekit/core/domain.py:14]: Should we check for these fields and throw an exception if they're not present?

Comment (NoB0) [reply dialoguekit/core/domain.py:14]: Good point! I added a check for the required fields.

File: dialoguekit/core/domain.py [hunk 2]
```diff
@@ -10,20 +10,40 @@ class Domain:
...
         if not os.path.isfile(config_file):
             raise FileNotFoundError(f"Config file not found: {config_file}")
+
         with open(config_file, encoding="utf-8") as yaml_file:
             self._config = yaml.load(yaml_file, Loader=yaml.FullLoader)
 
+        if not {"name", "slot_names"}.issubset(self._config):
+            raise KeyError(
+                "The domain configuration should contain at least the fields "
+                "name and slot_names."
```
Comment (kbalog) [inline dialoguekit/core/domain.py:34]: Nit: enclose field names in '...'

**Review by kbalog — APPROVED** [review]
LGTM

File: dialoguekit/core/domain.py (hunk 2 above)
Comment (NoB0) [reply dialoguekit/core/domain.py:34]: Done.

**Pull request merged**